Show remaining attendee capacity

nikomakela opened the pull request:
HH-449.

Bases on https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/866.

1. Replace custom schema generation script with built-in features.
2.  Update GraphQL schema types to include new registration field in event fields. Remove `remainingAttendeeCapacity` from event fields root.
3. Split EventHero component to multiple sub components.
4. Move keywords below event description in event details hero.
5. Use AppConfig to handle apps' include-parameter of event queries.
6. Disable registration / buy button when there is no space left in event.
7. Show "event full" -status in app's own event card (in search results list)

<img width="1284" height="1080" alt="image" src="https://github.com/user-attachments/assets/91ecd8b2-42c2-407d-b176-c946b4a2aa1e" />

<img width="1266" height="528" alt="image" src="https://github.com/user-attachments/assets/e401c0af-bfa3-42fd-b314-3e67ae82345c" />

nikomakela committed:
feat(components): add registration number fields to event details

HH-449.

nikomakela committed:
fix(codegen): graphql schema type generation using custom script

HH-449.

To fix GraphQL schema code generation with `codegen`, change file name
back to "pascalCaseWithUnderscoreBeforeLeadingDigit.js". The "mjs" file
format is not necessary and does not work with current setup, so the
easiest fix is to change the file format back to "js". Then `yarn
generate:graphql` works again.

nikomakela committed:
refactor(codegen): replace custom enum handling with built-in function

HH-449.

Remove custom script to handle enumerations when running codegen. It can
be replaced with some built in configuration.

nikomakela committed:
feat(apps,components): update graphql schema types

HH-449.

Add registration field to event details type. Misc updates to CMS and
Venue related hooks and types updated as a side effect.

nikomakela committed:
fix(apps,components): registration field changes

HH-449.

NOTE: registration field of event details is `null` if query params does
not contain `include=registration`.

nikomakela committed:
fix: remove needless and dangerous console logging

nikomakela committed:
feat(hobbies): add registration to event search "include" -query param

HH-449.

Add `include=registration` to main search and similar events query
params.

nikomakela committed:
refactor(components): split EventHero component

HH-449.

Split the EventHero component to multiple private function based
components, for sake of clarity.

nikomakela committed:
refactor(apps,components): move event hero keywords below short-desc

HH-449.

nikomakela committed:
refactor(hobbies): use AppConfig for include-param of event queries

HH-449.

nikomakela committed:
refactor(events): use AppConfig for include-param of event queries

HH-449.

nikomakela committed:
refactor(sports): use AppConfig for include-param of event queries

HH-449.

nikomakela committed:
feat: disable buy/enrol button in event page if remaining capacity is 0

HH-449

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr867.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
fix: remove link arrow label / enrolment status from carousel cards

HH-449.

The enrolmetn status should no longer be used as a link arrow label,
because instead of that, it has it's own definition item (icon + value)
in card next to others.

nikomakela committed:
refactor: remove event enrolment closed explanation

HH-449

nikomakela committed:
feat: add queueable status to enrolment status handling

HH-449

nikomakela committed:
feat: show event enrolment status also when event is enrollable

HH-449.

The event enrolment status indicators should also show some kind of
"event enrollable" text, when registration is ongoing.

nikomakela committed:
feat: always show event time in event details hero

HH-449.

Event times were not shown in the event hero component when super event
time range was same as the time of the detailed event. This commit
changes it show that the time is always shown also in the hero.

nikomakela committed:
feat: show event enrolment status as own new icon in event card

HH-449.

Instead of showing the event enrolmetn status next to link arrow, it is
now shown as a own new definition item next to others (icon + value).

nikomakela committed:
feat: event enrolment status

HH-449.

Add enrolment status info to event hero. Show only when enrolments are
activated for the event.

nikomakela committed:
refactor: move event hero keywords as last item in hero

HH-449.

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr867.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
refactor: use early returns in event hero sub components

HH-449

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr867.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
feat: add new event hero button text for queueable event

HH-449.

Add a new text to buy / enrol button when the enrolment status for event
is "queueable".

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr867.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
style: use a color of brick as alert color of enrolment status

HH-449.

Use a consistent color that is one of our primary colors in our app.
Contrast should be good enough as well.

nikomakela committed:
fix: event hero enrolment status indicators

HH-449

test: event hero enrolment features

HH-449.

WIP

WIP

WIP

WIP

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr867.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr867.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders commented on packages/components/src/components/eventHero/eventHero.module.scss:
```diff
@@ -57,7 +57,7 @@ $logoWidth: 5.5rem;
 
     .additionalInfo {
-      order: 6;
+      // order: 6;
```
Why not just remove this?

karisal-anders reviewed (approved):
Didn't try this out, just a fast code overview.

One thing came to mind:
- Translation of `enrollable` could be made more similar between languages, now they differ:
  - "Registration / Ilmoittautuminen käynnissä / Registrering"

Otherwise looks ok to merge, this should be tested by the product owner when in dev environment. Can be iterated more if needed, if product owner finds something to fix/change in dev environment.

nikomakela committed:
style: remove commented out line from css

HH-449.

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr867.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
feat: show event enrolment button before enrolment has started

HH-449.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr867.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
chore: upgrade react-helsinki-headless-cms to 1.1.5

HH-449.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr867.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
style(sonar): fix (trivial) security hot spot in tested url

HH-449.

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr867.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr867.dev.hel.ninja :laughing::tada::tada::tada:
